### PR TITLE
init ynh_setup_source

### PR DESCRIPTION
**Pull request by Thovi98**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Thovi98** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -34,20 +46,14 @@ ynh_add_nginx_config
...
+pushd $install_dir
     ynh_use_nodejs
 
 	ynh_script_progression --message="npm install... This can be very long, be patient !" --weight=18
-	ynh_exec_warn_less env $ynh_node_load_PATH $ynh_npm install --network-timeout=100000
+	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm ci --quiet --production --prefer-offline --network-timeout=100000
 
 	ynh_script_progression --message="npm run build... This can be very long, be patient !" --weight=25
-	ynh_exec_warn_less env $ynh_node_load_PATH $ynh_npm run build
+	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm run build
```
Comment (alexAubin) [inline scripts/install:56]:
```suggestion
	export SOURCE_VERSION="$(cat YNH_APP_BASEDIR/manifest.toml | toml_to_json | jq -r ".resources.sources.main.url" | awk -F/ '{ print $NF }')"
	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm run build
```

Looking at https://github.com/terraforming-mars/terraforming-mars/blob/main/make_static_json.js#L54, the build will assume it's running inside a git repo unless we define SOURCE_VERSION as the current commit, which we can obtain by parsing the manifest.toml

**Comment by alexAubin** [comment]
!testme

File: scripts/install [hunk 2]
```diff
@@ -34,20 +46,15 @@ ynh_add_nginx_config
...
     ynh_use_nodejs
 
 	ynh_script_progression --message="npm install... This can be very long, be patient !" --weight=18
-	ynh_exec_warn_less env $ynh_node_load_PATH $ynh_npm install --network-timeout=100000
+	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm ci --quiet --production --prefer-offline --network-timeout=100000
 
 	ynh_script_progression --message="npm run build... This can be very long, be patient !" --weight=25
-	ynh_exec_warn_less env $ynh_node_load_PATH $ynh_npm run build
+	export SOURCE_VERSION="$(cat $YNH_APP_BASEDIR/manifest.toml | toml_to_json | jq -r ".resources.sources.main.url" | awk -F/ '{ print $NF }')"
+	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $SOURCE_VERSION $ynh_npm run build
```
Comment (alexAubin) [inline scripts/install:57]:
```suggestion
	SOURCE_VERSION="$(cat $YNH_APP_BASEDIR/manifest.toml | toml_to_json | jq -r ".resources.sources.main.url" | awk -F/ '{ print $NF }')"
	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH SOURCE_VERSION="$SOURCE_VERSION" $ynh_npm run build
```

File: scripts/install [hunk 3]
```diff
@@ -34,20 +46,15 @@ ynh_add_nginx_config
...
-cd $install_dir/terraforming-mars
-ynh_exec_warn_less git checkout $TM_version
-
-pushd $install_dir/terraforming-mars
+pushd $install_dir
     ynh_use_nodejs
 
 	ynh_script_progression --message="npm install... This can be very long, be patient !" --weight=18
-	ynh_exec_warn_less env $ynh_node_load_PATH $ynh_npm install --network-timeout=100000
+	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm ci --quiet --production --prefer-offline --network-timeout=100000
```
Comment (alexAubin) [inline scripts/install:53]:
```suggestion
	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm ci --quiet --production --prefer-offline --network-timeout=100000
	ynh_exec_warn_less ynh_exec_as $app env $ynh_node_load_PATH $ynh_npm install --save-dev @types/html-escaper
```

**Comment by Thovi98** [comment]
@alexAubin You rock!

I still have to finish to cleanup the transition to `ynh_setup_source` but thanks for your help here!

**Comment by Thovi98** [comment]
!testme

**Pull request merged**